MessengerOnFileForLinux/MessageFramework/test/MessageTest.cpp

```diff
@@ -1,11 +1,36 @@
 #include <MessageTestCore.hpp>
+#include <regex>
+#include <cstring>
+#include <iostream>
 
-TEST_F(MessageFixture, isMessageCorrectCreatedFromFullMessageInRow)
+TEST_F(MessageFixture, isMessageCorrectCreatedFromParameters)
 {
-    EXPECT_TRUE(true);
+         const std::regex regexReadMessage("\\[0]\\[20([0-9]{2})-([0-9]{2})-([0-9]{2}) ([0-9]{2}):([0-9]{2}):([0-9]{2})]\\[mnurzyns]\\[test read message]");
+      const std::regex regexInviterMessage("\\[1]\\[20([0-9]{2})-([0-9]{2})-([0-9]{2}) ([0-9]{2}):([0-9]{2}):([0-9]{2})]\\[mnurzyns]\\[test inviter message]");
+    const std::regex regexRecipientMessage("\\[2]\\[20([0-9]{2})-([0-9]{2})-([0-9]{2}) ([0-9]{2}):([0-9]{2}):([0-9]{2})]\\[mnurzyns]\\[test recipient message]");
```

Comment:
nie widzę sensu testowania 3 przypadków. Jeden wystaczy. Co innego jakby sam Message w jakiś sposób ustawiał flagę, ale jest ona podawana w argumencie. Więc skoro działa dla argumentu "0" to tak samo zadziala dla arguemntu "1"

Reply:
Nope, nope, nope. 
>     setMessageFlag(messageFlag);
To w c-torze z parametrami, a dalej masz:

> bool Message::setMessageFlag(std::string messageFlag)
> {
>     log_.function("Message::setMessageFlag()");
>     messageFlag_ = messageFlag;
>     if(MessageFlag::inviterMessage == messageFlag || MessageFlag::recipientMessage == messageFlag)
>     {
>         std::string logData = "Message::setMessageFlag() messageFlag_ set as " + messageFlag;
>         log_.info(logData);
>         return true;
>     }
>     else if (MessageFlag::readMessage == messageFlag)
>     {
>         log_.info("Message::setMessageFlag() message read, can not unread");
>         return false;
>     }
>     else
>     {
>         log_.info("Message::setMessageFlag() messageFlag unknown, error");
>         return false;
>     }
> }

Reply:
Ok. To zrób z tego test parametryczny,

Reply:
Ok, zaakceptuj mi tego pull requesta. A w następnym zmienie wszystkie testy by były parametryczne i tylko wywoływały jakąś inną funkcję.